src/util/makeClickable.ts

```diff
@@ -0,0 +1,46 @@
+import { forceUpdate } from '@stencil/core';
+
+export function makeClickable(host) {
+    let isActive;
+    let hasJustReleasedEnter;
+
+    if (isActive === undefined || hasJustReleasedEnter === undefined) {
+        isActive = false;
+        hasJustReleasedEnter = true;
+    }
+
+    host.addEventListener('keydown', (event: KeyboardEvent) => {
+        if (event.key === 'Enter' && !isActive) {
+            isActive = true;
+
+            forceUpdate(host);
+        }
+    });
+
+    host.addEventListener('keyup', (event: KeyboardEvent) => {
+        if (event.key === 'Enter' && isActive) {
+            isActive = false;
+            hasJustReleasedEnter = true;
+
+            forceUpdate(host);
+        }
+    });
+
+    host.addEventListener(
+        'click',
+        (event: MouseEvent) => {
+            if (!isActive) {
+                return;
+            }
+
+            if (hasJustReleasedEnter) {
+                hasJustReleasedEnter = false;
+
+                return;
+            }
+
+            event.stopImmediatePropagation();
+        },
+        true
+    );
+}
```

Comment:
We should probably allow for some cleanup as well, if a consumer no longer wants the element to have this behavior. Now, once you have called this function you are stuck with the behavior forever.